9.1.0 and 9.1.1 no longer able to access anonymous S3 bucket like 9.0.3 and prior

**Describe the bug**
With CyberDuck v9.1.1, I am not able to connect to this S3 bucket, but with CyberDuck v9.0.3 and prior, I am.  It seems something has broke in the later versions.  Nothing obvious happens when I try to connect - there are no errors or other messages in the program.

**bucket address:**
`s3.amazonaws.com/gov-fpac-rma-pubfs-production`

![image](https://github.com/user-attachments/assets/08e6ac2a-7b93-42db-a98c-f116ab851ed4)

**versions I tested:**
v8.6.0 (39818) - works
v8.8.2 (41344) - works
v9.0.3 (42112) - works
v9.1.0 (42520) - broke
v9.1.1 - broke

**Exception in the debug log:**
```
2025-01-03 12:16:16,203 [background-5] WARN  ch.cyberduck.core.threading.BackgroundCallable - Failure running background task WorkerBackgroundAction{worker=MountWorker{cache=Cache{size=0}}}
java.lang.Exception: null
	at ch.cyberduck.core.threading.BackgroundCallable.<init>(BackgroundCallable.java:38) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.threading.DefaultBackgroundExecutor.execute(DefaultBackgroundExecutor.java:71) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.AbstractController.background(AbstractController.java:72) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at cli.Ch.Cyberduck.Ui.Controller.BrowserController$$$003C$$003Ec__DisplayClass229_0.<UnmountImpl>b__0(Unknown Source) ~[Cyberduck.exe:?]
	at ch.cyberduck.core.threading.BackgroundCallable$1.run(BackgroundCallable.java:64) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at cli.System.Delegate.DynamicInvokeImpl(Unknown Source) ~[mscorlib.dll:?]
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallbackDo(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallbackHelper(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Threading.ExecutionContext.RunInternal(Unknown Source) ~[mscorlib.dll:?]
	at cli.System.Threading.ExecutionContext.Run(Unknown Source) ~[mscorlib.dll:?]
	at cli.System.Threading.ExecutionContext.Run(Unknown Source) ~[mscorlib.dll:?]
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallback(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Control.InvokeMarshaledCallbacks(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Control.WndProc(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Form.WndProc(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.NativeWindow.Callback(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.UnsafeNativeMethods.DispatchMessageW(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Application$ComponentManager.System.Windows.Forms.UnsafeNativeMethods.IMsoComponentManager.FPushMessageLoop(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Application$ThreadContext.RunMessageLoopInner(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.System.Windows.Forms.Application$ThreadContext.RunMessageLoop(Unknown Source) ~[System.Windows.Forms.dll:?]
	at cli.Ch.Cyberduck.Ui.Program.Main(Unknown Source) ~[Cyberduck.exe:?]
Caused by: java.lang.NullPointerException
	at ch.cyberduck.core.auth.AWSCredentialsConfigurator$2.getAWSAccessKeyId(AWSCredentialsConfigurator.java:112) ~[Cyberduck.Protocols.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at com.amazonaws.auth.AbstractAWSSigner.sanitizeCredentials(AbstractAWSSigner.java:438) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.auth.AWS4Signer.sign(AWS4Signer.java:227) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1349) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1183) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:838) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:805) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:779) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:735) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:717) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:579) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:559) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.doInvoke(AWSSecurityTokenServiceClient.java:1731) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.invoke(AWSSecurityTokenServiceClient.java:1698) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.invoke(AWSSecurityTokenServiceClient.java:1687) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.executeGetCallerIdentity(AWSSecurityTokenServiceClient.java:1298) ~[Cyberduck.Protocols.DLL:?]
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.getCallerIdentity(AWSSecurityTokenServiceClient.java:1266) ~[Cyberduck.Protocols.DLL:?]
	at ch.cyberduck.core.s3.S3Session.login(S3Session.java:347) ~[Cyberduck.Protocols.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.KeychainLoginService.authenticate(KeychainLoginService.java:169) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.LoginConnectionService.authenticate(LoginConnectionService.java:163) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.LoginConnectionService.connect(LoginConnectionService.java:154) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.LoginConnectionService.check(LoginConnectionService.java:99) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.pool.StatelessSessionPool.borrow(StatelessSessionPool.java:57) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.threading.SessionBackgroundAction.run(SessionBackgroundAction.java:115) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.threading.SessionBackgroundAction$1.call(SessionBackgroundAction.java:106) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.threading.DefaultRetryCallable.call(DefaultRetryCallable.java:52) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.threading.SessionBackgroundAction.call(SessionBackgroundAction.java:108) ~[Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.threading.BackgroundCallable.run(BackgroundCallable.java:102) [Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at ch.cyberduck.core.threading.BackgroundCallable.call(BackgroundCallable.java:84) [Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:625) [?:1.8.0]
	at ch.cyberduck.core.threading.NamedThreadFactory$1.run(NamedThreadFactory.java:57) [Cyberduck.Core.DLL:de7c4824a60166cce2202a8f80e8cf7a43c9b2e0]
	at java.lang.Thread.run(Thread.java:955) [?:1.8.0]
```

**To Reproduce**
Steps to reproduce the behavior:
1. Open CyberDuck 9.1.1
2. Configure a connection as the screenshot above shows
3. Double click that connection to attempt to connect to it.
4. Notice that nothing happens.  A brief text change appears in the bottom status bar and the top title bar, but it disappears before I can read it.
5. Now, repeat these steps with [version 8.6.0](https://cyberduck.en.uptodown.com/windows/versions), and it will connect fine and show you all the files in the bucket.

**Expected behavior**
The current version should connect to the S3 bucket and show the files like v9.0.3 and prior did.

**Desktop (please complete the following information):**
 - OS: Windows 11 Pro 24H2
 - Version: CyberDuck 9.1.1

**Log Files**
[cyberduck 9.1.1 snipped.log](https://github.com/user-attachments/files/18302056/cyberduck.9.1.1.snipped.log)

--- a/Changelog.txt
+++ b/Changelog.txt
@@ -4,6 +4,7 @@ https://cyberduck.io/
 9.1.2
 - [Bugfix] No fallback to IPv4 address when IPv6 connect attempt fails with unreachable network (#16723)
 - [Bugfix] File → Rename option always disabled (#16695)
+- [Bugfix] Connect anonymously to public buckets (S3) (#16746)
 
 9.1.1
 - [Bugfix] Repeated prompts for passphrase (Cryptomator, Windows) (#16645)

--- a/core/src/main/java/ch/cyberduck/core/Credentials.java
+++ b/core/src/main/java/ch/cyberduck/core/Credentials.java
@@ -58,11 +58,6 @@ public class Credentials implements Comparable<Credentials> {
     private boolean saved = new LoginOptions().save;
 
     /**
-     * Passed authentication successfully
-     */
-    private boolean passed;
-
-    /**
      * Default credentials
      */
     public Credentials() {
@@ -78,7 +73,6 @@ public class Credentials implements Comparable<Credentials> {
         this.identityPassphrase = copy.identityPassphrase;
         this.certificate = copy.certificate;
         this.saved = copy.saved;
-        this.passed = copy.passed;
     }
 
     public Credentials(final String user) {
@@ -109,7 +103,6 @@ public class Credentials implements Comparable<Credentials> {
 
     public void setUsername(final String user) {
         this.user = user;
-        this.passed = false;
     }
 
     public Credentials withUsername(final String user) {
@@ -131,7 +124,6 @@ public class Credentials implements Comparable<Credentials> {
 
     public void setPassword(final String password) {
         this.password = password;
-        this.passed = false;
     }
 
     public Credentials withPassword(final String password) {
@@ -145,7 +137,6 @@ public class Credentials implements Comparable<Credentials> {
 
     public void setToken(final String token) {
         this.tokens = new TemporaryAccessTokens(token);
-        this.passed = false;
     }
 
     public Credentials withToken(final String token) {
@@ -159,7 +150,6 @@ public class Credentials implements Comparable<Credentials> {
 
     public void setTokens(final TemporaryAccessTokens tokens) {
         this.tokens = tokens;
-        this.passed = false;
     }
 
     public Credentials withTokens(final TemporaryAccessTokens tokens) {
@@ -201,14 +191,6 @@ public class Credentials implements Comparable<Credentials> {
         return this;
     }
 
-    public boolean isPassed() {
-        return passed;
-    }
-
-    public void setPassed(final boolean passed) {
-        this.passed = passed;
-    }
-
     /**
      * @return true if the username is anonymous.
      */
@@ -244,7 +226,6 @@ public class Credentials implements Comparable<Credentials> {
 
     public Credentials withIdentity(final Local file) {
         this.identity = file;
-        this.passed = false;
         return this;
     }
 
@@ -262,7 +243,6 @@ public class Credentials implements Comparable<Credentials> {
      */
     public void setIdentity(final Local file) {
         this.identity = file;
-        this.passed = false;
     }
 
     public String getIdentityPassphrase() {

--- a/core/src/main/java/ch/cyberduck/core/Credentials.java
+++ b/core/src/main/java/ch/cyberduck/core/Credentials.java
@@ -213,10 +213,7 @@ public class Credentials implements Comparable<Credentials> {
      * @return true if the username is anonymous.
      */
     public boolean isAnonymousLogin() {
-        if(StringUtils.isEmpty(user)) {
-            return false;
-        }
-        return PreferencesFactory.get().getProperty("connection.login.anon.name").equals(user);
+        return StringUtils.equals(user, PreferencesFactory.get().getProperty("connection.login.anon.name"));
     }
 
     public boolean isPasswordAuthentication() {

--- a/core/src/main/java/ch/cyberduck/core/KeychainLoginService.java
+++ b/core/src/main/java/ch/cyberduck/core/KeychainLoginService.java
@@ -174,7 +174,6 @@ public class KeychainLoginService implements LoginService {
         catch(LoginFailureException e) {
             log.debug("Login failed for {}", session);
             listener.message(LocaleFactory.localizedString("Login failed", "Credentials"));
-            credentials.setPassed(false);
             final LoginOptions options = new LoginOptions(bookmark.getProtocol());
             if(this.prompt(bookmark, e.getDetail(), prompt, options)) {
                 // Retry
@@ -204,8 +203,6 @@ public class KeychainLoginService implements LoginService {
         else {
             log.info("Skip writing credentials for bookmark {}", bookmark.getHostname());
         }
-        // Flag for successful authentication
-        credentials.setPassed(true);
         // Nullify password and tokens
         log.debug("Reset credentials for {}", bookmark);
         switch(bookmark.getProtocol().getStatefulness()) {

--- a/profiles/default/S3 (HTTPS).cyberduckprofile
+++ b/profiles/default/S3 (HTTPS).cyberduckprofile
@@ -27,6 +27,8 @@
         <string>s3.amazonaws.com</string>
         <key>Hostname Configurable</key>
         <true/>
+        <key>Anonymous Configurable</key>
+        <true/>
         <key>Port Configurable</key>
         <false/>
         <key>Default Nickname</key>

--- a/s3/src/main/java/ch/cyberduck/core/cloudfront/CloudFrontDistributionConfiguration.java
+++ b/s3/src/main/java/ch/cyberduck/core/cloudfront/CloudFrontDistributionConfiguration.java
@@ -615,8 +615,10 @@ public class CloudFrontDistributionConfiguration implements DistributionConfigur
 
     private AmazonCloudFront client(final Path container) throws BackgroundException {
         final AmazonCloudFrontClientBuilder builder = AmazonCloudFrontClientBuilder.standard()
-            .withCredentials(AWSCredentialsConfigurator.toAWSCredentialsProvider(session.getClient().getProviderCredentials()))
             .withClientConfiguration(configuration);
+        if(session.getClient().isAuthenticatedConnection()) {
+            builder.withCredentials(AWSCredentialsConfigurator.toAWSCredentialsProvider(session.getClient().getProviderCredentials()));
+        }
         final Location.Name region = this.getRegion(container);
         if(S3Session.isAwsHostname(session.getHost().getHostname(), false)) {
             if(Location.unknown.equals(region)) {

--- a/s3/src/main/java/ch/cyberduck/core/kms/KMSEncryptionFeature.java
+++ b/s3/src/main/java/ch/cyberduck/core/kms/KMSEncryptionFeature.java
@@ -130,8 +130,10 @@ public class KMSEncryptionFeature extends S3EncryptionFeature {
 
     private AWSKMS client(final Path container) throws BackgroundException {
         final AWSKMSClientBuilder builder = AWSKMSClientBuilder.standard()
-                .withCredentials(AWSCredentialsConfigurator.toAWSCredentialsProvider(session.getClient().getProviderCredentials()))
                 .withClientConfiguration(configuration);
+        if(session.getClient().isAuthenticatedConnection()) {
+            builder.withCredentials(AWSCredentialsConfigurator.toAWSCredentialsProvider(session.getClient().getProviderCredentials()));
+        }
         final Location.Name region = location.getLocation(container);
         if(S3Session.isAwsHostname(session.getHost().getHostname(), false)) {
             if(Location.unknown.equals(region)) {

--- a/s3/src/main/java/ch/cyberduck/core/restore/Glacier.java
+++ b/s3/src/main/java/ch/cyberduck/core/restore/Glacier.java
@@ -109,6 +109,9 @@ public class Glacier implements Restore {
         final AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard()
             .withCredentials(AWSCredentialsConfigurator.toAWSCredentialsProvider(session.getClient().getProviderCredentials()))
             .withClientConfiguration(configuration);
+        if(session.getClient().isAuthenticatedConnection()) {
+            builder.withCredentials(AWSCredentialsConfigurator.toAWSCredentialsProvider(session.getClient().getProviderCredentials()));
+        }
         final Location.Name region = this.getRegion(container);
         if(S3Session.isAwsHostname(session.getHost().getHostname(), false)) {
             if(Location.unknown.equals(region)) {

--- a/s3/src/main/java/ch/cyberduck/core/s3/S3Session.java
+++ b/s3/src/main/java/ch/cyberduck/core/s3/S3Session.java
@@ -312,62 +312,55 @@ public class S3Session extends HttpSession<RequestEntityRestStorageService> {
     public void login(final LoginCallback prompt, final CancelCallback cancel) throws BackgroundException {
         final Credentials credentials = authentication.get();
         if(credentials.isAnonymousLogin()) {
-            log.debug("Connect with no credentials to {}", host);
+            log.debug("Connect non-authenticated with no credentials for {}", host);
             client.setProviderCredentials(null);
         }
         else {
             if(credentials.getTokens().validate()) {
-                log.debug("Connect with session credentials to {}", host);
+                log.debug("Connect with session token for {}", host);
                 client.setProviderCredentials(new AWSSessionCredentials(
                         credentials.getTokens().getAccessKeyId(), credentials.getTokens().getSecretAccessKey(),
                         credentials.getTokens().getSessionToken()));
             }
             else {
-                log.debug("Connect with basic credentials to {}", host);
+                log.debug("Connect with static keys for {}", host);
                 client.setProviderCredentials(new AWSCredentials(credentials.getUsername(), credentials.getPassword()));
             }
         }
         final Path home = new DelegatingHomeFeature(new DefaultPathHomeFeature(host)).find();
         if(S3Session.isAwsHostname(host.getHostname(), false)) {
             if(StringUtils.isEmpty(RequestEntityRestStorageService.findBucketInHostname(host))) {
-                final CustomClientConfiguration configuration = new CustomClientConfiguration(host,
-                        new ThreadLocalHostnameDelegatingTrustManager(trust, host.getHostname()), key);
-                final AWSSecurityTokenServiceClientBuilder builder = AWSSecurityTokenServiceClientBuilder.standard()
-                        .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(host.getProtocol().getSTSEndpoint(), null))
-                        .withCredentials(AWSCredentialsConfigurator.toAWSCredentialsProvider(client.getProviderCredentials()))
-                        .withClientConfiguration(configuration);
-                final AWSSecurityTokenService service = builder.build();
-                // Returns details about the IAM user or role whose credentials are used to call the operation.
-                // No permissions are required to perform this operation.
-                try {
-                    final GetCallerIdentityResult identity = service.getCallerIdentity(new GetCallerIdentityRequest());
-                    log.debug("Successfully verified credentials for {}", identity);
-                }
-                catch(AWSSecurityTokenServiceException e) {
-                    throw new STSExceptionMappingService().map(e);
-                }
-            }
-            else {
-                final Location.Name location = new S3LocationFeature(this, regions).getLocation(home);
-                log.debug("Retrieved region {}", location);
-                if(!Location.unknown.equals(location)) {
-                    log.debug("Set default region to {} determined from {}", location, home);
-                    host.setProperty("s3.location", location.getIdentifier());
+                if(client.isAuthenticatedConnection()) {
+                    final CustomClientConfiguration configuration = new CustomClientConfiguration(host,
+                            new ThreadLocalHostnameDelegatingTrustManager(trust, host.getHostname()), key);
+                    final AWSSecurityTokenServiceClientBuilder builder = AWSSecurityTokenServiceClientBuilder.standard()
+                            .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(host.getProtocol().getSTSEndpoint(), null))
+                            .withCredentials(AWSCredentialsConfigurator.toAWSCredentialsProvider(client.getProviderCredentials()))
+                            .withClientConfiguration(configuration);
+                    final AWSSecurityTokenService service = builder.build();
+                    // Returns details about the IAM user or role whose credentials are used to call the operation.
+                    // No permissions are required to perform this operation.
+                    try {
+                        final GetCallerIdentityResult identity = service.getCallerIdentity(new GetCallerIdentityRequest());
+                        log.debug("Successfully verified credentials for {}", identity);
+                        return;
+                    }
+                    catch(AWSSecurityTokenServiceException e) {
+                        throw new STSExceptionMappingService().map(e);
+                    }
                 }
             }
         }
+        if(home.isRoot() && StringUtils.isEmpty(RequestEntityRestStorageService.findBucketInHostname(host))) {
+            log.debug("Skip querying region for {}", home);
+            new S3ListService(this, acl).list(home, new DisabledListProgressListener());
+        }
         else {
-            if(home.isRoot() && StringUtils.isEmpty(RequestEntityRestStorageService.findBucketInHostname(host))) {
-                log.debug("Skip querying region for {}", home);
-                new S3ListService(this, acl).list(home, new DisabledListProgressListener());
-            }
-            else {
-                final Location.Name location = new S3LocationFeature(this, regions).getLocation(home);
-                log.debug("Retrieved region {}", location);
-                if(!Location.unknown.equals(location)) {
-                    log.debug("Set default region to {} determined from {}", location, home);
-                    host.setProperty("s3.location", location.getIdentifier());
-                }
+            final Location.Name location = new S3LocationFeature(this, regions).getLocation(home);
+            log.debug("Retrieved region {}", location);
+            if(!Location.unknown.equals(location)) {
+                log.debug("Set default region to {} determined from {}", location, home);
+                host.setProperty("s3.location", location.getIdentifier());
             }
         }
     }

--- a/s3/src/main/java/ch/cyberduck/core/s3/S3Session.java
+++ b/s3/src/main/java/ch/cyberduck/core/s3/S3Session.java
@@ -327,10 +327,6 @@ public class S3Session extends HttpSession<RequestEntityRestStorageService> {
                 client.setProviderCredentials(new AWSCredentials(credentials.getUsername(), credentials.getPassword()));
             }
         }
-        if(host.getCredentials().isPassed()) {
-            log.warn("Skip verifying credentials with previous successful authentication event for {}", this);
-            return;
-        }
         final Path home = new DelegatingHomeFeature(new DefaultPathHomeFeature(host)).find();
         if(S3Session.isAwsHostname(host.getHostname(), false)) {
             if(StringUtils.isEmpty(RequestEntityRestStorageService.findBucketInHostname(host))) {

--- a/webdav/src/main/java/ch/cyberduck/core/dav/DAVSession.java
+++ b/webdav/src/main/java/ch/cyberduck/core/dav/DAVSession.java
@@ -158,10 +158,6 @@ public class DAVSession extends HttpSession<DAVClient> {
                 client.disablePreemptiveAuthentication();
             }
         }
-        if(credentials.isPassed()) {
-            log.warn("Skip verifying credentials with previous successful authentication event for {}", this);
-            return;
-        }
         try {
             final Path home = this.getFeature(Home.class).find();
             final HttpHead head = new HttpHead(new DAVPathEncoder().encode(home));

--- a/www/update/changelog.html
+++ b/www/update/changelog.html
@@ -94,6 +94,9 @@
     <li><span class="label label-warning">Bugfix</span> File → Rename option always disabled (<a
             target="_blank" href="https://trac.cyberduck.io/ticket/16695">#16695</a>)
     </li>
+    <li><span class="label label-warning">Bugfix</span> Connect anonymously to public buckets (S3) (<a
+            target="_blank" href="https://trac.cyberduck.io/ticket/16746">#16746</a>)
+    </li>
 </ul>
 
 <p>

